src/obfuscation/hidden.c

```diff
@@ -0,0 +1,123 @@
+#include "hidden.h"
+#include "rmalloc.h"
+#include "util/minmax.h"
+#include "redis_index.h"
+#include "query_node.h"
+#include "reply_macros.h"
+
+typedef struct {
+  const char *user;
+  uint64_t length;
+} UserString;
+
+HiddenName *NewHiddenName(const char* name, uint64_t length, bool takeOwnership) {
+  UserString* value = rm_malloc(sizeof(*value));
+  value->user = name;
+  if (takeOwnership) {
+    value->user = rm_strndup(name, length);
+  }
```

Comment:
Consider avoiding the write-after-write? (use an else clause?)
Do we know which branch would be taken more often? (use `__builtin_expect`?)